components/AirdropLogin.vue

```diff
@@ -108,22 +109,39 @@
   </div>
 </template>
 <script lang="ts">
-import { Vue, Component, Prop } from 'vue-property-decorator'
-import { CONNECT_WALLET_TYPES } from '~/constant'
+import { Vue, Component, Prop, Watch } from 'vue-property-decorator'
+import {
+  CONNECT_WALLET_TYPES,
+  COSMOS_ADDRESS_REGEX,
+  OSMO_ADDRESS_REGEX,
+} from '~/constant'
 
 @Component
 export default class AirdropLogin extends Vue {
   @Prop(Boolean) readonly isAirdropStarted: boolean | undefined
 
   inputAddress: string = ''
+  errorMessage: string = ''
 
   // eslint-disable-next-line class-methods-use-this
   get connectWalletTypes() {
     return CONNECT_WALLET_TYPES
   }
 
+  @Watch('inputAddress')
+  onValueChange() {
+    this.errorMessage = ''
+  }
+
   handleAddressInput() {
-    this.$emit('input', this.inputAddress)
+    this.errorMessage = ''
+    if (
+      COSMOS_ADDRESS_REGEX.test(this.inputAddress) ||
+      OSMO_ADDRESS_REGEX.test(this.inputAddress)
+    ) {
+      this.$emit('input', this.inputAddress)
+    }
+    this.errorMessage = this.$t('AirDrop.errorMessage.address') as string
```

Comment:
Missing return, even if this passes the test, `this.errorMessage` would still be set
Also prefer guarding for error using i, and emit normal case at the end of function, rather than checking for valid case using if, and throwing error as default